### DISCUSSION TITLE
Automatic the process of creating sub-accounts (multi-chain account)

### DISCUSSION THREAD

**Discussion by AlexiJemano** [phantom/docs#246] [General]
Hello 

My question is how to automate this proces let's say I'm building an app (like salad) where every user gets their own address. Doing it manually will be hard so if I could write a py code using the API and let's say connect to main account and say to create 1000 sub-accounts. How I do it?

**Comment by adamdelphantom** [comment]
Hello @AlexiJemano,

The Phantom provider does not provide any functionality for creating accounts. To do this you will want to construct your transactions using a library like [Solana's Web3js](https://solana-labs.github.io/solana-web3.js/). I am not aware of any libraries that allow you to build a accounts for multiple chains.

**Reply by AlexiJemano** [reply]
I will look in to it but I got a response from someone else that says that the sub-account (multi-chain account) - Quote _

> "The manner by which Phantom presents onchain wallets and in-app “user accounts” is just organizational. Accounts aren’t really multichain, they are just bundled that way. The single piece of data that all wallets in one “account” share is one passphrase of words. The passphrase generates the first private key and wallet address on each blockchain. From there on, on each blockchain, the next wallet is created from the previous wallet data. So, if I understand your question correctly, you just iterate the wallet creation step for a single blockchain for as many users you need."

_

So I made a pyhton code to create them:
For Bitcoin:
`from bip_utils import Bip39SeedGenerator, Bip44, Bip44Coins, Bip44Changes, Bip39MnemonicGenerator

def generate_mnemonic():
    return Bip39MnemonicGenerator().FromWordsNumber(12)

def derive_btc_wallets_from_seed(seed_phrase, num_wallets):
    seed_bytes = Bip39SeedGenerator(seed_phrase).Generate()
    bip44_mst_ctx = Bip44.FromSeed(seed_bytes, Bip44Coins.BITCOIN)
    wallets = []
    for i in range(num_wallets):
        bip44_acc_ctx = bip44_mst_ctx.Purpose().Coin().Account(0).Change(Bip44Changes.CHAIN_EXT).AddressIndex(i)
        wallets.append({
            "address": bip44_acc_ctx.PublicKey().ToAddress(),
            "private_key": bip44_acc_ctx.PrivateKey().Raw().ToHex(),
            "public_key": bip44_acc_ctx.PublicKey().RawCompressed().ToHex()
        })
    return wallets

def save_wallets_to_file(wallets, filename):
    with open(filename, 'w') as file:
        for wallet in wallets:
            file.write(f"{wallet['address']},{wallet['private_key']},{wallet['public_key']}\n")

if __name__ == "__main__":
    master_mnemonic = generate_mnemonic()
    print(f"Master Mnemonic: {master_mnemonic}")
    num_wallets = 1000
    btc_wallets = derive_btc_wallets_from_seed(master_mnemonic, num_wallets)
    save_wallets_to_file(btc_wallets, 'btc_wallets.txt')
    print(f"Created {num_wallets} BTC wallets and saved to btc_wallets.txt")
`

For ETH:
`from bip_utils import Bip39SeedGenerator, Bip44, Bip44Coins, Bip44Changes, Bip39MnemonicGenerator

def generate_mnemonic():
    return Bip39MnemonicGenerator().FromWordsNumber(12)

def derive_eth_wallets_from_seed(seed_phrase, num_wallets):
    seed_bytes = Bip39SeedGenerator(seed_phrase).Generate()
    bip44_mst_ctx = Bip44.FromSeed(seed_bytes, Bip44Coins.ETHEREUM)
    wallets = []
    for i in range(num_wallets):
        bip44_acc_ctx = bip44_mst_ctx.Purpose().Coin().Account(0).Change(Bip44Changes.CHAIN_EXT).AddressIndex(i)
        wallets.append({
            "address": bip44_acc_ctx.PublicKey().ToAddress(),
            "private_key": bip44_acc_ctx.PrivateKey().Raw().ToHex(),
            "public_key": bip44_acc_ctx.PublicKey().RawCompressed().ToHex()
        })
    return wallets

def save_wallets_to_file(wallets, filename):
    with open(filename, 'w') as file:
        for wallet in wallets:
            file.write(f"{wallet['address']},{wallet['private_key']},{wallet['public_key']}\n")

if __name__ == "__main__":
    master_mnemonic = generate_mnemonic()
    print(f"Master Mnemonic: {master_mnemonic}")
    num_wallets = 1000
    eth_wallets = derive_eth_wallets_from_seed(master_mnemonic, num_wallets)
    save_wallets_to_file(eth_wallets, 'eth_wallets.txt')
    print(f"Created {num_wallets} ETH wallets and saved to eth_wallets.txt")
`

I tried to make for SOL and for POLY but it dosen't really work so for now I will use only BTC and ETH

**Reply by adamdelphantom** [reply]
@AlexiJemano The response you provided is mostly correct. And we do not support a way to programmatically create or import accounts into Phantom. Any wallets generated outside of Phantom will need to be imported via the ui by the user.
